cpp/src/jni/dataset/jni_util.cc

```diff
@@ -16,36 +16,44 @@
 // under the License.
 
 #include "jni/dataset/jni_util.h"
-
+#include "arrow/ipc/metadata_internal.h"
+#include "arrow/util/base64.h"
+#include "arrow/util/key_value_metadata.h"
 #include "arrow/util/logging.h"
 
+#include <memory>
 #include <mutex>
 
+#include <flatbuffers/flatbuffers.h>
+
 namespace arrow {
+
+namespace flatbuf = org::apache::arrow::flatbuf;
+
 namespace dataset {
 namespace jni {
 
 class ReservationListenableMemoryPool::Impl {
  public:
-  explicit Impl(arrow::MemoryPool* pool, std::shared_ptr<ReservationListener> listener,
+  explicit Impl(MemoryPool* pool, std::shared_ptr<ReservationListener> listener,
                 int64_t block_size)
       : pool_(pool),
         listener_(listener),
         block_size_(block_size),
         blocks_reserved_(0),
         bytes_reserved_(0) {}
 
-  arrow::Status Allocate(int64_t size, uint8_t** out) {
```

Comment:
please try to avoid these types of cleanups with large functional changes.

Reply:
Sorry about that. :( Will make split changes next time.

Reply:
I agree with @emkornfield. To split these types of changes into another PR make us easier to review this PR.

Reply:
Hello @kiszk , this part of code (Cross JNI data sharing) was moved to https://github.com/apache/arrow/pull/10883 as a dependency of this PR, would you like to have a look? Thank a lot. I have added a separated style fix commit in that one.

Reply:
Or you think we should have a individual PR rather than a commit to fix the style stuffs? I am okay to either way.

Reply:
Sure, let me take a look.